docs/dev/reader-concurrency-semaphore.md

```diff
@@ -47,19 +47,19 @@ The dump contains the following information:
 * Dump of the permit states;
 
 Permits are grouped by table, description, and state, while groups are sorted by memory consumption.
-The first group in this example contains 34 permits, all for reads against table `ks1.table1_mv_0`, all data-query reads and in state `active/blocked`.
+The first group in this example contains 34 permits, all for reads against table `ks1.table1_mv_0`, all data-query reads and in state `active/await`.
 
 Permits have the following states:
 * waiting - the permit is waiting for admission;
-* active/unused - the permit was admitted but doesn't participate in CPU based admission;
-* active/used - the permit was admitted and it participates in CPU based admission;
-* active/blocked - a previously active/used permit, which needs something other than CPU to proceed, it is waiting on I/O or a remote shards;
+* active - the permit was admitted;
+* active/need_cpu - the permit was admitted and it participates in CPU based admission;
```

Comment:
What does "admitted and participates in CPU admission mean"?
It's hard to understand without reading the code.
When does "admitted" mean?
How is is different from "waiting" (pre-admission state)?
Can the state machine be drawn here (using basic ASCII art)?

Reply:
It is explained above and I didn't repeat it here. I may need to expand the explanation.

Reply:
I don't think a state-machine with ASCII art is feasible. Only with an image.